kubernetes/provider.go

```diff
@@ -364,3 +365,49 @@ func initializeConfiguration(d *schema.ResourceData) (*restclient.Config, error)
 	log.Printf("[INFO] Successfully initialized config")
 	return cfg, nil
 }
+
+var useadmissionregistrationv1beta1 *bool
+
+func useAdmissionregistrationV1beta1(conn *kubernetes.Clientset) (bool, error) {
+	if useadmissionregistrationv1beta1 != nil {
+		return *useadmissionregistrationv1beta1, nil
+	}
+
+	d := conn.Discovery()
+	groups, err := d.ServerGroups()
+	if err != nil {
+		return false, err
+	}
+
+	apiGroup := "admissionregistration.k8s.io"
+	use, err := func() (bool, error) {
```

Comment:
This anon function is a bit confusing. 
Why not just have `use` and `err` scoped to `useAdmissionregistrationV1beta1` set them and break out of the loop with `continue` instead of `return`?